src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -369,6 +369,19 @@ def get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood=None, release
         raise CLIInternalError("No content was found in helm registry path response.")
 
 
+def send_request_with_retries(cli_ctx, method, url, headers, fault_type, summary, uri_parameters=None, resource=None, retry_count=5, retry_delay=3):
+    for i in range(retry_count):
+        try:
+            response = send_raw_request(cli_ctx, method, url, headers=headers, uri_parameters=uri_parameters, resource=resource)
+            if response.status_code == 200:
+                return response
+        except Exception as e:
+            if i == retry_count - 1:
+                telemetry.set_exception(exception=e, fault_type=fault_type, summary=summary)
+                raise CLIInternalError("Error while fetching helm chart registry path: " + str(e))
+            time.sleep(retry_delay)
```

Comment:
delay should be outside try/except block

Reply:
for non 200 response, flow reaches exception block and hence the sleep can stay inside the block